src/averageAges.js

```diff
@@ -36,8 +51,20 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenFilter = people
+    .filter(({ sex }) => sex === SEX_FEMALE);
+
+  const womenFilterWithChild = womenFilter
+    .filter(({ name }) => people.some(({ mother }) => mother === name));
+
+  const selectedWomen = withChildren ? womenFilterWithChild : womenFilter;
```

Comment:
Now everything is written correctly, but if `withChildren` is `false`, then you are doing unnecessary filtering. It is better to first check if this value is `true` and then if it is necessary to do a new filtering.